### PR TITLE
WIP 178 hide quote button for non-player.

### PR DESCRIPTION
**Pull request by mikelax**
This was a start to hide the Quote buttons for non-players. 

@nazar  I have a question on this. Currently we are only looking at if the user is logged in or not in the `_viewingControls` function where we set the `canPost` attribute. 

This also needs to check if the user is a player in the game. 
In `ViewGameMessages`, I was going to pass a new property of `myGamePlayer={myGamePlayer}` to `GameMessages`. I then realized though that this prop would then have to be passed down to: 

MessagesRenderer -> GameMessageContainer -> GameMessage

I am not sure if this is the "right" approach, plus I wasn't sure if that was the right approach given the props we may need to track for #177

### CODE REVIEW

File: client/src/components/Games/View/ViewGameMessages.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ function ViewGameMessages(props) {
     <React.Fragment>
       <GameHeader game={game} />
 
-      <GamesMessages gameId={game.id}/>
+      <GameMessages gameId={_.toInteger(game.id)}/>
```
Comment (mikelax) [inline client/src/components/Games/View/ViewGameMessages.js]: @nazar  is the `toInteger` correct, or should the "id" attribute already be a number within the game object?

Comment (nazar) [reply client/src/components/Games/View/ViewGameMessages.js]: `game.id` is defined in the graphQL as an `ID` scalar type, which a string.

**Comment by mikelax** [comment]
thanks for the work @nazar . I was going to ask about changing the proptype to `string` i guess that is what we should do for props that are scalars. 

With regards to the other refactor, I see what you did, would another option have been to use the new context API? It must have been late as I didn't think of that last night, until this morning while walking the dog.

File: client/src/components/Games/View/ViewGameLoungeMessages.js [hunk 2]
```diff
@@ -19,15 +21,36 @@ function GameLoungeMessagesView (props) {
       <GameLoungeMessages gameId={id}/>
 
       <Segment>
-        <CreateLoungeMessage gameId={id}/>
+        { isAuthenticated && <CreateLoungeMessage gameId={id}/> }
```
Comment (mikelax) [inline client/src/components/Games/View/ViewGameLoungeMessages.js:24]: @nazar  what do you think about this pattern of using the `isAuthenticated` variable for determining the component to display? 

other option would be to do an if check in the render function before the return and assign the correct component to a variable.

Comment (nazar) [reply client/src/components/Games/View/ViewGameLoungeMessages.js:24]: >what do you think about this pattern of using the isAuthenticated variable for determining the component to display?

It's perfect 👍 💯

**Review by nazar — APPROVED** [review]
🚀

**Pull request merged**